tools/federation/src/main/java/org/eclipse/rdf4j/federated/algebra/PrecompiledQueryNode.java

```diff
@@ -0,0 +1,43 @@
+/*******************************************************************************
+ * Copyright (c) 2019 Eclipse RDF4J contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *******************************************************************************/
+package org.eclipse.rdf4j.federated.algebra;
+
+import org.eclipse.rdf4j.federated.endpoint.provider.SailSourceEvaluationStrategyFactory;
+import org.eclipse.rdf4j.federated.evaluation.SailTripleSource;
+import org.eclipse.rdf4j.query.algebra.QueryRoot;
+import org.eclipse.rdf4j.query.algebra.TupleExpr;
+
+/**
+ * Node representing a precompiled query.
+ * 
+ * @author Andreas Schwarte
+ * @see SailSourceEvaluationStrategyFactory
+ * @see SailTripleSource
+ */
+public class PrecompiledQueryNode extends QueryRoot {
```

Comment:
Is it necessary for this class to be public?

Reply:
This class is used in two different sub packages (evaluation and endpoint.provider). I currently do not see a good way of restricting visibility. Since this is a simple agebra node I however do not have that much of a problem having this public